docs/ARCHITECTURE.md

```
@@ -0,0 +1,715 @@
+# Architecture Overview
+
+## High-Level Design
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)